src/main/scala/com/mongodb/spark/sql/fields/MaxKey.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spark.sql.fields
+
+import org.bson.BsonMaxKey
+
+/**
+ * The MinKey companion object
+ *
+ * @since 1.0
+ */
+object MaxKey {
+  /**
+   * Create a new instance
+   *
+   * @return the new instance
+   */
+  def apply(): MaxKey = new MaxKey(1)
+}
+
+/**
+ * A case class representing the Bson MaxKey type
+ *
+ * @param maxKey data representing the maxKey
+ * @since 1.0
+ */
+case class MaxKey(maxKey: Int) extends Field[BsonMaxKey] {
```

Comment:
Because the Struct has to have a field.  

The extend json representation of a MaxKey is: `{ $maxKey: 1}`  so the struct mirrors that with a schema of `maxKey: Int`.  

The case class / java bean conversion in Spark requires getter and setters for the fields of the struct.